app/Http/Controllers/Api/AssetsController.php

```diff
@@ -746,7 +746,21 @@ public function restore(Request $request, $assetId = null)
         return response()->json(Helper::formatStandardApiResponse('error', null, trans('admin/hardware/message.does_not_exist')), 200);
     }
 
-
+    /**
+     * Checkout an asset by its tag.
+     *
+     * @author [N. Butler]
+     * @param string $tag
+     * @since [v6.0.5]
+     * @return JsonResponse
+     */
+    public function checkoutByTag(AssetCheckoutRequest $request, $tag)
+    {
+        if ($asset = Asset::with('assetstatus')->with('assignedTo')->where('asset_tag', $tag)->first()) {
```

Comment:
Do we actually need the `with('assetstatus')->with('assignedTo')` stuff? The `checkout()` method already tries to determine if we can check the item out or not.